src/main/scala/DebuggableDetails.scala

```diff
@@ -45,6 +45,9 @@ object DebuggableTestDetails extends App {
         Debuggable(result, s"h: input: $a, result: $result")
     }
 
+    //I couldn't understand this. How can the 'for' extract an integer (e.g. fRes) from the State Monad?
+    //by means of 'map'? Sorry to bother with these dummy questions but I have experienced only fors which
+    //have worked with Lists :)
     val finalResult = for {
```

Comment:
If you translate the for <- to a flatMap you get:

> _f(100).flatMap(fResult => g(fResult).flatMap..._

I would expect fResult were a _Debuggable_ BUT is an Int... How can the 'for' know how to get it?